### PR TITLE
Ensure that the SemaphoreStep always gets its TimingAction & test this

### PR DESCRIPTION
**Pull request by svanoort**
Solves [JENKINS-38737](https://issues.jenkins-ci.org/browse/JENKINS-38737). 

Possibly a bit hacky, but the step is only used for testing. 

@reviewbybees especially @jglick

### CODE REVIEW

File: src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/SemaphoreTimingActionTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.test.steps.SemaphoreStep;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class SemaphoreTimingActionTest {
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/SemaphoreTimingActionTest.java]: Better to call it simply `SemaphoreStepTest`.

File: src/test/java/org/jenkinsci/plugins/workflow/test/steps/SemaphoreStep.java [hunk 2]
```diff
@@ -152,7 +155,14 @@ public static void waitForStart(@Nonnull String k, @CheckForNull Run<?,?> b) thr
 
         @Inject(optional=true) private SemaphoreStep step;
 
+        @StepContextParameter
+        private transient FlowNode node;
+
         @Override public boolean start() throws Exception {
+            // Because
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/test/steps/SemaphoreStep.java]: Because what?

**Comment by svanoort** [comment]
@jglick The issue seems to be fairly specific to Semaphore implementation in this case, ergo a fix here

**Comment by jglick** [comment]
> The issue seems to be fairly specific to Semaphore implementation in this case

But what is the root cause?

**Comment by svanoort** [comment]
@jglick In this case, it appeared to be interference between the Semaphore GraphListener and the one in WorkflowRun.

**Comment by jglick** [comment]
Well that [seems to be unused](https://github.com/search?utf8=%E2%9C%93&q=user%3Ajenkinsci+SemaphoreListener&type=Code&ref=searchresults) so I am not sure how it could hurt anything, though it ought to be deleted. What brought you to this conclusion?

**Comment by svanoort** [comment]
@jglick The fix was tested and did indeed resolve the issue.  However it's been weeks so I don't remember the investigation details in depth.   IMO the fix here was sufficient for a test-only class where I could have simply done a workaround and moved on.   But if you want to dig deeper, feel free to pick up the issue yourself as the resident expert: https://issues.jenkins-ci.org/browse/JENKINS-38737  -- it's unassigned now.

**Comment by jglick** [comment]
OK.

**Pull request closed without merging**